service/scaladsl/kafka/server/src/test/scala/com/lightbend/lagom/scaladsl/kafka/broker/ScaladslKafkaApiSpec.scala

```diff
@@ -217,6 +215,25 @@ class ScaladslKafkaApiSpec extends WordSpecLike with Matchers with BeforeAndAfte
       for (i <- 1 to batchSize) queue.offer((i.toString, NoOffset))
       latch.await(10, TimeUnit.SECONDS) shouldBe true
     }
+
+    "allow the consumer to batch (using eventually)" in {
+      val batchSize = 4
+      val counter = new AtomicInteger(0)
+      testService.test7Topic.subscribe.atLeastOnce(
+        Flow[String].grouped(batchSize).mapConcat { messages =>
+          messages.map { _ =>
+            counter.incrementAndGet()
+            Done
+          }
+        }
+      )
+      val queue = ScaladslKafkaApiSpec.test7Queue.futureValue
+      for (i <- 1 to batchSize) queue.offer((i.toString, NoOffset))
+      eventually {
+        counter.intValue() shouldBe batchSize
+      }
+    }
```

Comment:
This is a rewrite of the batch test (see above) which avoids using `await` countdownLatch and uses an `AtomicInteger` and `eventually` instead.

I haven't seen it fail in 20 runs on my machine.

I'm keeping both to run a small experiment on Travis to see which is more stable.